iot/api/CloudIotTest/CloudIotTest.cs

```diff
@@ -0,0 +1,334 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+using Google.Cloud.Iam.V1;
+using Google.Cloud.PubSub.V1;
+using Grpc.Core;
+using Xunit;
+using System;
+
+namespace GoogleCloudSamples
+{
+    // <summary>
+    /// Runs the sample app's methods and tests the outputs
+    // </summary>
+    public class CommonTests
+    {
+        private static readonly string s_projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+        private static readonly string s_regionId = "us-central1";
+        private static readonly string s_serviceAccount = "serviceAccount:cloud-iot@system.gserviceaccount.com";
+
+
+        readonly CommandLineRunner _cloudIot = new CommandLineRunner()
+        {
+            VoidMain = CloudIotSample.Main,
+            Command = "CloudIotSample"
+        };
+
+        protected ConsoleOutput Run(params string[] args)
+        {
+            return _cloudIot.Run(args);
+        }
+
+        private readonly RetryRobot _retryRobot = new RetryRobot()
+        {
+            RetryWhenExceptions = new[] { typeof(Xunit.Sdk.XunitException) }
+        };
+
+        private void Eventually(Action action) => _retryRobot.Eventually(action);
+
+        private void CreatePubSubTopic(String name)
+        {
+            var publisher = PublisherClient.Create();
+            TopicName topicName = new TopicName(s_projectId, name);
+            try
+            {
+                publisher.CreateTopic(topicName);
+            }
+            catch (RpcException e)
+            when (e.Status.StatusCode == StatusCode.AlreadyExists)
+            {
+            }
+            Policy policy = new Policy
+            {
+                Bindings =
+                    {
+                        new Binding {
+                            Role = "roles/pubsub.publisher",
+                            Members = { s_serviceAccount }
+                        }
+                    }
+            };
+            SetIamPolicyRequest request = new SetIamPolicyRequest
+            {
+                Resource = $"projects/{s_projectId}/topics/{topicName.TopicId}",
+                Policy = policy
+            };
+            Policy response = publisher.SetIamPolicy(request);
+            Console.WriteLine($"Topic IAM Policy updated: {response}");
+        }
+
+        private void DeletePubSubTopic(String name)
+        {
+            PublisherClient publisher = PublisherClient.Create();
+            TopicName topicName = new TopicName(s_projectId, name);
+            publisher.DeleteTopic(topicName);
+        }
+
+        [Fact]
+        public void TestCreateDeleteRegistry()
+        {
+            var registryId = "testcreatereg";

```

Comment:
It can be a problem if two people will run the same set of tests in parallel using the same project Id. We should add a degree of uniqueness to object names if we use real GCP APIs and not mocks.

Reply:
+1 -- Done.  The only potential issue before with collisions was if the teardown failed, the tests would potentially fail on next run.  Using part of a GUID should prevent this.